app/src/main/java/com/github/multimatum_team/multimatum/AddDeadlineActivity.kt

```diff
@@ -1,57 +1,103 @@
 package com.github.multimatum_team.multimatum
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.DatePickerDialog
+import android.app.DatePickerDialog.OnDateSetListener
+import android.app.TimePickerDialog
 import android.os.Bundle
 import android.view.View
-import android.widget.DatePicker
 import android.widget.TextView
 import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
 import com.github.multimatum_team.multimatum.model.Deadline
 import com.github.multimatum_team.multimatum.model.DeadlineState
-import com.github.multimatum_team.multimatum.repository.DeadlineRepository
+import com.github.multimatum_team.multimatum.viewmodel.DeadlineListViewModel
 import dagger.hilt.android.AndroidEntryPoint
-import kotlinx.coroutines.runBlocking
-import java.time.LocalDate
+import java.time.Instant
 import java.time.LocalDateTime
-import javax.inject.Inject
+import java.time.ZoneId
+import java.time.temporal.ChronoUnit
+
 
 /**
- *  Activity who create a deadline using a DatePicker
+ *  Activity who create a deadline using a DatePickerDialog and a TimePickerDialog
  */
 @AndroidEntryPoint
 class AddDeadlineActivity : AppCompatActivity() {
 
-    companion object {
-        // Temporary value that will be link later to the actual day
-        private val initDate : LocalDateTime = LocalDateTime.of(2022,1,1, 0, 0)
-    }
+    private var selectedDate: LocalDateTime =
```

Comment:
You should use `clockService.now()` instead of `Instant.now().atZone(ZoneId.systemDefault())` and thus inject clockService to the class. Then you should mock a clockService for your tests.

I think using `now()`without specifying the zoneId will use the systemDefault one by default.